<?xml version="1.0"?>
<robot xmlns:xacro="http://www.ros.org/wiki/xacro">

  <xacro:macro name="itri_arm_ros2_control" params="name initial_positions_file">

    <ros2_control name="${name}" type="system">
      <hardware>
        <plugin>itri_arm_hardware/ItriArmHardware</plugin>
        <param name="robot_ip">192.168.1.10</param>
        <param name="robot_port">4000</param>
        <param name="target_ip">192.168.1.20</param>
        <param name="read_frequency">100.0</param>
      </hardware>
      
      <joint name="joint_1">
        <command_interface name="position">
          <param name="min">-3.14159</param>
          <param name="max">3.14159</param>
        </command_interface>
        <state_interface name="position"/>
        <state_interface name="velocity"/>
        <state_interface name="effort"/>
      </joint>
      
      <joint name="joint_2">
        <command_interface name="position">
          <param name="min">-3.14159</param>
          <param name="max">3.14159</param>
        </command_interface>
        <state_interface name="position"/>
        <state_interface name="velocity"/>
        <state_interface name="effort"/>
      </joint>
      
      <joint name="joint_3">
        <command_interface name="position">
          <param name="min">-3.14159</param>
          <param name="max">3.14159</param>
        </command_interface>
        <state_interface name="position"/>
        <state_interface name="velocity"/>
        <state_interface name="effort"/>
      </joint>
      
      <joint name="joint_4">
        <command_interface name="position">
          <param name="min">-3.14159</param>
          <param name="max">3.14159</param>
        </command_interface>
        <state_interface name="position"/>
        <state_interface name="velocity"/>
        <state_interface name="effort"/>
      </joint>
      
      <joint name="joint_5">
        <command_interface name="position">
          <param name="min">-3.14159</param>
          <param name="max">3.14159</param>
        </command_interface>
        <state_interface name="position"/>
        <state_interface name="velocity"/>
        <state_interface name="effort"/>
      </joint>
      
      <joint name="joint_6">
        <command_interface name="position">
          <param name="min">-3.14159</param>
          <param name="max">3.14159</param>
        </command_interface>
        <state_interface name="position"/>
        <state_interface name="velocity"/>
        <state_interface name="effort"/>
      </joint>

    </ros2_control>

  </xacro:macro>

</robot>
